### PR TITLE
feat: Add support for o1-preview and o1-mini on the OpenAI provider

### PR DESCRIPTION
**Pull request by Christopher-Hayes**
- **:art: Fix: Update versions of @ai-sdk/azure and @ai-sdk/openai**
- **:sparkles: Feat: Add support for o1 reasoning models**

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `ModelOption` component for selecting conversation models, enhancing user interface options.
	- Added new models (`o1`, `o1-preview`, `o1-mini`) to the selection list, expanding available choices for users.

- **Improvements**
	- Updated model management logic to handle "thinking models" differently, optimizing the initial message construction based on selected models.
	- Enhanced dependency versions for improved performance and stability.

- **Bug Fixes**
	- Adjusted token handling for specific models during text generation, ensuring accurate behavior based on model type.

### CODE REVIEW

File: src/renderer/components/ModelOption.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  currentlySelectedId,
+  vscode,
+  showParentMenu,
+  currentConversation,
+  setShowModels,
+}: {
+  currentConversation: Conversation;
+  model: RichModel;
+  currentlySelectedId?: string;
+  vscode: any;
```
[the remark this answers is not in the document]
Comment (Christopher-Hayes) [reply src/renderer/components/ModelOption.tsx:21]: Assigning a type to vscode causes too many issues.

**Comment by PeterDaveHello** [comment]
Should o1 be added right now? I thought it's not ready yet?

**Comment by Christopher-Hayes** [comment]
@PeterDaveHello right now it'll work like a normal `gpt-4o` model. The ChatGPT UI will show a GUI for what it's "thinking" as it runs. We can't show that yet, so visually it works like any other model, and does not show streaming. But otherwise it works.

The GUI updates are delayed due to the fact the Vercel `ai` package we use doesn't fully support reasoning models yet. But, we're moving to LangChang anyways, so we may be able to add those GUI updates sooner once we've switched.

**Comment by Christopher-Hayes** [comment]
Based on the original issue I'll look at creating issues for enhancements specific to reasoning models.

**Comment by Christopher-Hayes** [comment]
o1 API availability is still tier-based, but in recent weeks it's become more available.

As of Oct. 1, tier 3-5 users are now able to use o1. https://x.com/OpenAIDevs/status/1841176573527077235

For tier 1-2 users, the extension _should_ hide the o1 models until they receive access.

**Pull request merged**